lib/Parse/Lexer.cpp

```diff
@@ -2404,6 +2401,11 @@ void Lexer::lexImpl() {
     return formToken(tok::eof, TokStart);
   }
 
+  // If we have experimental string processing enabled, try to lex a regex
+  // literal.
+  if (tryLexRegexLiteral(TokStart))
+    return;
```

Comment:
Doing this for every token might cause noticeable performance regression. Can we move this to
`case '#':` and `case 'r'`?

Reply:
Fair enough, how does this look?